post-til-virksomhet/src/main/java/no/difi/meldingsutveksling/ptv/CorrespondenceAgencyMessageFactory.java

```diff
@@ -13,147 +14,189 @@
+        binaryAttachmentV2.setData(reporteeFactory.createBinaryAttachmentV2Data(new DataHandler(InputStreamDataSource.of(fileEntry.getInputStream()))));
+        return binaryAttachmentV2;
+    }
 
-        no.altinn.services.serviceengine.reporteeelementlist._2010._10.ObjectFactory reporteeFactory = new no.altinn.services.serviceengine.reporteeelementlist._2010._10.ObjectFactory();
-        BinaryAttachmentExternalBEV2List attachmentExternalBEV2List = new BinaryAttachmentExternalBEV2List();
+    private FileEntryStream getFileEntry(String messageId, BusinessMessageFile f) {
         try {
-            List<NoarkDocument> noarkDocuments = PayloadUtil.parsePayloadForDocuments(edu.getPayload());
-            noarkDocuments.forEach(d -> {
-                BinaryAttachmentV2 binaryAttachmentV2 = new BinaryAttachmentV2();
-                binaryAttachmentV2.setFunctionType(AttachmentFunctionType.fromValue("Unspecified"));
-                binaryAttachmentV2.setFileName(reporteeFactory.createBinaryAttachmentV2FileName(d.getFilename()));
-                binaryAttachmentV2.setName(reporteeFactory.createBinaryAttachmentV2Name(d.getFilename()));
-                binaryAttachmentV2.setEncrypted(false);
-                binaryAttachmentV2.setSendersReference(reporteeFactory.createBinaryAttachmentV2SendersReference("AttachmentReference_as123452"));
-                binaryAttachmentV2.setData(reporteeFactory.createBinaryAttachmentV2Data(Base64.getDecoder().decode(d.getContent())));
-                attachmentExternalBEV2List.getBinaryAttachmentV2().add(binaryAttachmentV2);
-            });
-
-            String title = PayloadUtil.queryPayload(edu.getPayload(), "Melding/journpost/jpInnhold");
-            String content = PayloadUtil.queryPayload(edu.getPayload(), "Melding/journpost/jpOffinnhold");
-
-            return create(config, edu.getId(), edu.getReceiver().getIdentifier(), title, content, attachmentExternalBEV2List);
-        } catch (PayloadException e) {
-            throw new MeldingsUtvekslingRuntimeException("Error querying payload for Dokument", e);
+            return cryptoMessagePersister.readStream(messageId, f.getIdentifier());
+        } catch (IOException e) {
+            throw new NextMoveRuntimeException(
+                    String.format("Could not read attachment %s for messageId = %s", f.getIdentifier(), messageId)
+                    , e);
         }
-
     }
 
-    public static InsertCorrespondenceV2 create(CorrespondenceAgencyConfiguration config,
-                                                String conversationId,
-                                                String receiverIdentifier,
-                                                String messageTitle,
-                                                String messageContent,
-                                                BinaryAttachmentExternalBEV2List attachments) {
+    public InsertCorrespondenceV2 create(String conversationId,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 8 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00107)